internal/api/impl/proxy/proxy.go

```diff
@@ -101,33 +47,26 @@ type Proxy struct {
 	Crypto       crypto.Crypto
 }
 
-func (e *Proxy) Proxy() echo.MiddlewareFunc {
-	return func(next echo.HandlerFunc) echo.HandlerFunc {
-		return func(c echo.Context) error {
-			requestPath := c.Request().URL.Path
-			globalDatasourceMatch := globalProxyMatcher.MatchString(requestPath)
-			projectDatasourceMatch := projectProxyMatcher.MatchString(requestPath)
-			dashboardDatasourceMatch := dashboardProxyMatcher.MatchString(requestPath)
-			if !globalDatasourceMatch && !projectDatasourceMatch && !dashboardDatasourceMatch {
-				// this is likely a request for the API itself
-				return next(c)
-			}
-			if globalDatasourceMatch {
-				return e.proxyGlobalDatasource(c)
-			}
-			if projectDatasourceMatch {
-				return e.proxyProjectDatasource(c)
-			}
-			return e.proxyDashboardDatasource(c)
-		}
+func New(dashboardDAO dashboard.DAO, secretDAO secret.DAO, globalSecretDAO globalsecret.DAO, dtsDAO datasource.DAO, globalDtsDAO globaldatasource.DAO, crypto crypto.Crypto) route.Endpoint {
+	return &endpoint{
+		Dashboard:    dashboardDAO,
+		Secret:       secretDAO,
+		GlobalSecret: globalSecretDAO,
+		DTS:          dtsDAO,
+		GlobalDTS:    globalDtsDAO,
+		Crypto:       crypto,
 	}
 }
 
-func (e *Proxy) proxyGlobalDatasource(ctx echo.Context) error {
-	dtsName, path, err := extractGlobalDatasourceAndPath(ctx.Request().URL.Path)
-	if err != nil {
-		return err
-	}
+func (e *endpoint) CollectRoutes(g *route.Group) {
+	g.ANY(fmt.Sprintf("/%s/:%s/*", utils.PathGlobalDatasource, utils.ParamName), e.proxyGlobalDatasource, true)
+	g.ANY(fmt.Sprintf("/%s/:%s/%s/:%s/*", utils.PathProject, utils.ParamProject, utils.PathDatasource, utils.ParamName), e.proxyProjectDatasource, true)
```

Comment:
probably at some point we would like to protect these routes with the permission `read` on the datasource.

But can be down later as it is not the case for the moment